### DISCUSSION TITLE
Thanks Patrick!

### DISCUSSION THREAD

**Discussion by oldgristmill** [mcguirepr89/BirdNET-Pi#215] [General]
Had some water damage to my microphone a few weeks ago so had to set this project aside until the new one arrived, and as I visited this repo for the first time a while, it looks like @mcguirepr89 is stepping back from supporting this project and this seems to be a recent decision. As someone who just found this project and has absolutely no useful coding skills and often needs a bit of hand-holding but is otherwise an enthusiastic supporter, I'm wondering what this means now for this project.

To Patrick specifically, I'd like to thank you for all the work you've put into this project and helping folks like me. You've not said anything about why you want to pass on the repo and take yourself out of the loop, but really... you don't need to. Just know it's appreciated.

To everyone else, I'd love to better understand how this project fits into the broader ecosystem of associated BirdNET projects... I know an improved version was being worked on by @mcguirepr89, and there are things like BirdNET-Analyzer-Pi. I might be able to recruit some development support if I can get a sense of what the intended roadmap is/was, even I'm absolutely useless for coding anything more complex than BASIC on my old Commodore64. :)

**Comment by ehpersonal38** [comment]
Yes, even though I only found the project about a week ago I was quite shocked to learn that he's stepping back from the project. I can see all the effort he put in over the last year or so, and it's really great how he's helped make birding more accessible and enjoyable to people.

I'll be continuing to try my best to support the project code-wise when I can (mainly with issues or enhancements), but as for the broader scope of the project going forward I'm not sure what he had planned. I do think he wanted to try and focus more on making the backend more accessible to external applications, like an api of sorts that could be used by more than just the web app.

**Comment by nilspupils** [comment]
Sadly, things are not looking good at all at the moment. If no solution can be found this will be the end of BirdNET-Pi as we know it. 
Patrick is such an integral part in this project that  simply transferring the repo to someone else will just not work. He is the heart and soul of BirdNET-Pi and absolutely irreplaceable in this project. 
The reason for this mess seems to be overly strict licensing rules to favor the commercial variants of BirdNET systems as can be seen from his last reply here: https://github.com/mcguirepr89/BirdNET-Pi/discussions/184#discussion-4009745
We all should support Patrick in finding a solution that allows him to continue his work on terms that value his contribution! 
Best, Nils

**Comment by Kestrelio** [comment]
Aw dang. I just found out about it a couple days ago too. It would be awesome if this project could continue in some fashion.

**Comment by filikun** [comment]
Is there something we as non coding fans can do? It was not that clear on why he stepped back and it seem to have come out of the blue as 10d ago you where having a discussion on the new model. 

_"As of now, I am not allowed to accept any sponsorship donations to aid in developing BirdNET-Pi around the new model."_

Is this the reason maybe?

I've never seen someone so passionate of a project, would be sad if it dies. Who can we make contact with to push this project to be able to get funding?

**Reply by mcguirepr89** [reply]
Hi, @filikun -- thank you for your kind sentiments and ambition to solve this dilemma.

I don't really understand the logic behind the decisions that have been made, but you are correct that Cornell (and HaikuBox) disallowing any form of voluntary public funding for this (Cornell-endorsed) project is the main obstacle in being able to sustain the project. 

I've put many months of hard work into this project, but Cornell does not want this project to take money away from HaikuBox (a company into which individuals at Cornell have personal monetary investments).

For this reason, you can certainly try to ask them to do the right thing, but that won't happen.

A sad American story indeed

**Reply by filikun** [reply]
> Cornell does not want this project to take money away from HaikuBox

First off thanks for clarifying. I do not understand how this project can take money away from a commercial product. They serve two completely different purposes will appeal two completely different customer groups. Even if you made the installation process brutally simple it still requires one to be some what tech savvy.

I see that BirdNET is under a NonCommercial license and for all means I'm no expert but my limited knowledge is that it does not prevent payment, it just prevents commercial purposes. Like if your intent is to sell your own hardware/software (like the HaikuBox) it will be against the license but getting some funding to continue the project is not commercial nor do you profit from it commercially.

I don't know if you where trying to get them to fund you or if you were asking if you where allowed to get the funding externally but you should be able to setup a "Buy me a coffee" or something at least without any red bells going off. I mean you are your own developer, one could fund you, not the project itself 😅

Very sad to hear it anyways. I do see your passion in how you reply to comments here on GitHub. And a project like this feels like it's just waiting to be broader known. Like I found it because someone in the Home Assistant (smart home open source project) community was talking about finding a way to somehow integrate Birdnet pi into Home Assistant. 

I would love to help on the project but I have really no programming skills outside of being a heavy smart home enthusiast with all the tinkering with hardware and software that comes with it. Maybe you could find people willing to help out with this project over at the Home Assistant community. 

Here's the link from the [Reddit post that got me interested in trying out Birdnet pi.](https://reddit.com/r/homeassistant/comments/u6nacb/birdnetpi_automated_locally_run_bird_sound/)

Sorry for the wall of text. I hope this somehow will work out for you @mcguirepr89, what I can tell you deserve the credit.

**Comment by CaiusX** [comment]
Another wall of text .......

I am in agreement with @filikun and others that this project should not shut down due to prohibition of  community backing and funding, so that as many BirdNET-Pi (combined with home assist/home weather stations/webcam installations etc) have a "plug and play" system creating a history of the "ecosystem health" of their backyard/farm/nature reserve/industrial facility/island/city/state/country and planet.  

I am sponsoring Patrick $10 a month [HERE](https://github.com/sponsors/mcguirepr89) for him to buy SD cards and other hardware for continued development (and maybe a  cup of coffee or Chinese Takeaway) and if that gets shut down, I will find another means to make sure that when I'm sitting at my desk and hear 

https://birds.naturestation.net/By_Date/2022-04-25/Cape_Robin-Chat/Cape_Robin-Chat-71-2022-04-25-birdnet-07:04:30.mp3

I can look at this 

![image](https://user-images.githubusercontent.com/25373276/165050238-a76bcd0e-eca0-4215-b9b5-59a340855da3.png)

and at this 

![image](https://user-images.githubusercontent.com/25373276/165050513-4eac40ba-8288-46c8-9b0d-45f72644e782.png)

grab my camera and go and take this

![image](https://user-images.githubusercontent.com/25373276/165050705-72a2ee1b-bf6c-4f7a-af22-00a6070ce730.png)

In addition, I commit to making all my labeled sound files available FREE OF CHARGE for academic researchers & developers to progress this exciting field.

With the sudden recent spike in traffic through this repository (325 stars and climbing), I would encourage anyone with an interest in keeping this open source project developing at its rapid pace to contribute either through technical assistance and/or financial contribution.

Yours in renewed hope for a better world

African Penguin
.

**Reply by DD4WH** [reply]
> From a lay perspective, the validity of the license change to "non-commercial" is suspect, considering BirdNET-Pi's development commencing before the license change - see my early October [interactions ](https://github.com/mcguirepr89/BirdNET-Pi/discussions/19#discussion-3616425).

That appears to be a valid point, since BirdNET-Lite has been licensed with the Apache license 2.0 stating:

" 2. Grant of Copyright License. Subject to the terms and conditions of
      this License, each Contributor hereby grants to You a perpetual,
      worldwide, non-exclusive, no-charge, royalty-free, **_irrevocable_**
      copyright license to reproduce, prepare Derivative Works of,
      publicly display, publicly perform, sublicense, and distribute the
      Work and such Derivative Works in Source or Object form."

Since the license is "irrevocable", changing the license later to include some "non-commercial" aspect is at least problematic.

**Reply by mcguirepr89** [reply]
Seems I'll need to respect the original license . . . sorry, Cornell :/

**Reply by mcguirepr89** [reply]
@kahst needs to respond

**Reply by DD4WH** [reply]
Yes, it is highly desirable that this topic comes to a solution that every contributor can live with !

**Comment by kahst** [comment]
Hey guys, let me quickly chime in here. We started publishing BirdNET models with an MIT license because we wanted BirdNET to be freely accessible to everyone with no restrictions. Unfortunately, we realized that people would exploit that and would start selling BirdNET analyses to other research groups (yes, that actually happened). This was something that we definitely never intended, especially if it was behind our backs. In order to avoid that in the future, we changed the license to "non-commercial" for all repos and decided that commercial projects would at least have to come to us and ask for a license (which we would grant with little to no royalties based on the size of the project). 

This is exactly what Loggerhead did for HaikuBox. But the license is very narrow and ONLY covers models that were trained using data that Loggerhead collected, if they are being used for consumer products. This does not affect running BirdNET on a Pi, or any other platform, this does not affect other models, this does not affect any spin-off that is not a commercial consumer product. So this DOES NOT affect BirdNET-Pi for two reasons: 1) BirdNET-Pi uses the BirdNET-Lite model which was NOT trained using Loggerhead's data, and 2) BirdNET-Pi is not a commercial consumer product. And @mcguirepr89 knows that (!), we met with David Mann and decided that Patrick can enable GitHub donations for BirdNET-Pi because of those two reasons. 

So, I honestly don't understand the grudge, I'm disappointed in the resentment I'm getting from this community, and I feel like we (the Lab and the Haikubox team) are being treated very unfairly in this thread. I appreciate all the work Patrick has put into BirdNET-Pi, I am a huge fan of the project and I would like to endorse and support it moving forward. But I won't continue to justify our license decision, it is what it is and we need to find a way to live with it.

Loggerhead and I agreed that we will only push models to the new BirdNET-Analyzer repo that were NOT trained using Loggerhead's data in order to avoid overlap with Haikubox and confusion with the non-commercial aspect of the license in the future (this will be the case with the upcoming upgrade to a 3K species model).

We can't expect everyone to like Haikubox, but I will ask you to acknowledge what Haikibox has done for BirdNET (the Lite and Analyzer repo) because the only reason why the published models are so efficient and still accurate is BECAUSE of our work for Haikubox.

I will live with whatever decision @mcguirepr89 makes, though it would be very unfortunate indeed if BirdNET-Pi were discontinued. But I also want to highlight that I am NOT to blame for that, neither is Loggerhead nor the Lab!

**Reply by filikun** [reply]
Thanks for clarifying. It sounds to me we are dealing with some kind of a misunderstanding? Or what kind of funding are you trying to get @mcguirepr89?

**Reply by mcguirepr89** [reply]
@filikun I wanted to enable github sponsorship on the repo for the new model around which I was asked to develop.

**Reply by kahst** [reply]
And that's perfectly fine, you can do that. However, you'll have to wait for the 3K model which I just started training.

**Reply by mcguirepr89** [reply]
Settled

**Reply by CaiusX** [reply]
@kahst, thank you for your side of the story and my obvious misreading of the licensing situation.  I have edited my original post deleting unfair comments.

Best
Dennis

**Reply by kahst** [reply]
Thanks, I appreciate that.

**Reply by nilspupils** [reply]
Hooray! Awesome this is settled! Thanks everyone for chiming in. Please sponsor @mcguirepr89 now! And please support the development of BirdNET, too!!!

**Comment by filikun** [comment]
So with that somewhat sorted I have two questions. 

1. I see that the Swedish database Is lacking a lot of translation, how can I help? 
2. Is there a Birdnet Pi discord (or the likes of it)? If not that's a great place for a community to grow, share and help the project.

**Reply by nilspupils** [reply]
Hi @filikun, in the swedish database are about 4 % translations missing. These are most likely from rare species, but it would be good if you could add some entries! You can find the file zipped in the model folder.

**Reply by filikun** [reply]
> Hi @filikun, in the swedish database are about 4 % translations missing. These are most likely from rare species, but it would be good if you could add some entries! You can find the file zipped in the model folder. 
> 
> 

Ah I see, I'll look into it. I have one species that is not translated everyday in my logs so I'm eager to get at least that one sorted 😅

**Reply by filikun** [reply]
> Hi @filikun, in the swedish database are about 4 % translations missing. These are most likely from rare species, but it would be good if you could add some entries! You can find the file zipped in the model folder.

I took a look at the `labels_sv.txt` and found my one species that where wrong. I guess the logic here is `latin-name_translated-name` and my specific species looked like this `Corvus monedula_Eurasian Jackdaw` but I can tell you that `Eurasian Jackdaw` is not Swedish. When creating these translation files does it fill it with English translations or something first? Is there anyway I can tell what strings are not translated and what are?

**Reply by CaiusX** [reply]
Hi @filikun

Thanks for your interest and initiative -I have opened a new https://github.com/mcguirepr89/BirdNET-Pi/discussions/224#discussion-4034527to consolidate in one place, please continue there else this important work may get lost?

Best
AP
